mono/mini/mini-arm64.c

```diff
@@ -5114,6 +5114,66 @@ emit_store_regset_cfa (MonoCompile *cfg, guint8 *code, guint64 regs, int basereg
 	return code;
 }
 
+/* Same as emit_load_regset, but emit unwind info too */
+/* and emit the load in reverse order to satisfy win32 unwind info */
+/* CFA_OFFSET is the offset between the CFA and basereg */
+static __attribute__((__warn_unused_result__)) guint8*
+emit_load_regset_cfa (MonoCompile* cfg, guint8* code, guint64 stored_regs, guint64 regs, int basereg, int offset, int cfa_offset, guint64 no_cfa_regset) {
+
+	int i, j, pos, nregs;
+	guint32 cfa_regset = regs & ~no_cfa_regset;
+	guint32 cfa_stored_regset = stored_regs & ~no_cfa_regset;
+
+	pos = -1;
+
+	for (i = 0; i < 32; ++i) {
+		if (cfa_stored_regset & (1 << i)) {
+			pos += 1;
+		}
+	}
+
+	for (i = 32; i >= 0; --i) {
+		nregs = 1;
+		if (cfa_regset & (1 << i)) {
+			if (pos > 0 && pos % 2 && (cfa_regset & (1 << (i - 1))) && (i - 1 != ARMREG_SP)) {
```

Comment:
This loop includes `i == 0`, do we need to worry about that case and `i = 1` expression becoming negative?

Reply:
Practically we should never hit that case since that's an arg resister and we shouldn't need to save it, but I've fixed that in https://github.com/Unity-Technologies/mono/pull/1960/commits/9b8e291ee6dde9c51871bf05de6ad195e1d48f01